tools/__init__.py

```diff
@@ -18,6 +18,41 @@
 
 
 def run_cmd(cmd, log_msg='', working_dir=None):
+    """Runs a command in the shell, raising an error if one occurs.
+
+    Args:
+        cmd (string): command to run
+        log_msg (string): purpose of the command
+        working_dir (string): location of arch_job_dir
+
+    Returns:
+        tuple of 3 elements containing
+        - stdout (string): stdout of the process
+        - stderr (string): stderr of the process
+        - exit_code (string): exit code of the process
+    """
+
+    stdout, stderr, exit_code = run_subprocess(cmd, log_msg, working_dir)
+
+    if exit_code != 0:
```

Comment:
Rather than breaking up the existing `run_cmd` function into two functions, and causing confusion on when to use which, how about adding a `raise_on_error` named argument to `run_cmd`, and letting that determine whether an error is only logged or it triggering the raising of a `RuntimeError`?